python/pyspark/sql/dataframe.py

```diff
@@ -3455,8 +3455,14 @@ def __getitem__(self, item: Union[int, str, Column, List, Tuple]) -> Union[Colum
             return self.filter(item)
         elif isinstance(item, (list, tuple)):
             return self.select(*item)
-        elif isinstance(item, int):
-            jc = self._jdf.apply(self.columns[item])
+        elif isinstance(item, int) and not isinstance(item, bool):
```

Comment:
Allowing `bool` is fine since `bool` inherits `int` in Python.

Reply:
ok, let me revert this part and add test cases for `bool`